src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -165,6 +166,19 @@ CorProfiler::Initialize(IUnknown* cor_profiler_info_unknown) {
     event_mask |= COR_PRF_DISABLE_OPTIMIZATIONS;
   }
 
+  const WSTRING domain_neutral_instrumentation =
+      GetEnvironmentValue(environment::domain_neutral_instrumentation);
+
+  if (domain_neutral_instrumentation == "1"_W || domain_neutral_instrumentation == "true"_W) {
+    Info("Detected environment variable ", environment::domain_neutral_instrumentation,
+         "=", domain_neutral_instrumentation);
+    Info("Enabling automatic instrumentation of methods called from domain-neutral assemblies. ",
+         "Please ensure that there is only one AppDomain or, if applications are being hosted in IIS, ",
+         "ensure that all Application Pools only have one application. ",
+         "Otherwise, a FileNotFound exception may occur.");
```

Comment:
Should we be more specific here and mention `HRESULT 0x80131401`  or `SECURITY_E_INCOMPATIBLE_SHARE`? [Reference](https://github.com/DataDog/dd-trace-dotnet/blob/9cfca307cc9253c04869fe2cb81fcb008893ce9f/lib/coreclr/src/inc/corerror.xml#L1293-L1297)

Reply:
Yes I think that's a good idea. I'll insert the HRESULT since that seems to result in better Google Search results.

Reply:
Resolved in 1a79b7b5239ab20af134e26484cca6db56e29cb5